test/Query.spec.js

```diff
@@ -1353,6 +1353,32 @@ describe('Query', () => {
         );
       });
 
+      it('supports field expression', async () => {
+        const query = new Query(User);
+        query.where(User.fields.id.notEqual(1));
+        await expect(
+          query.fetch(),
+          'to be fulfilled with sorted rows satisfying',
+          [new User({ id: 2, name: 'User 2' })]
+        );
+      });
+
+      it('supports multiple field expression', async () => {
+        const query = new Query(User);
+        const where = new Query.Where();
+        query.where(
+          where.or(User.fields.id.equal(1), User.fields.name.equal('User 2'))
+        );
+        await expect(
+          query.fetch(),
+          'to be fulfilled with sorted rows satisfying',
+          [
+            new User({ id: 1, name: 'User 1' }),
+            new User({ id: 2, name: 'User 2' })
+          ]
+        );
+      });
+
```

Comment:
We need some more tests here. Off the top of my mind:

- `in`, with an array: `Model.fields.id.in([1, 2])`
- `in`, with varargs: `Model.fields.id.in(1, 2)` (i think this should work, not sure though)
- `between`, with an array: `Model.fields.id.between([1, 3])`
- `between`, with varargs: `Model.fields.id.between(1, 3)`
- any option, with raw SQL: `Model.fields.name.equals(query.sql(`trim('User 2')`))`
- any option, with parameterized raw SQL: `Model.fields.name.equals(query.sql('trim($1)', '    User 1   '))`
- an `and` wrapper, similar to the one you have now for `where.or`
- a `not` wrapper, similar to the one you have now for `where.or`
- a couple of tests for `having`: `query.groupBy('id').having(User.fields.age.equal(10))`;

Reply:
Thanks for suggesting the tests. I am never used to writing tests, its something that I need to get myself acquainted with.

Reply:
Sure thing. Though these tests I suggest here though are very much a copy-paste effort from Query ;)

Reply:
@joelmukuthu I am sorry I couldn't push the tests. I added it a while back but my laptop had some issues and got it back this week only. And also 3 tests fail right now

Reply:
@joelmukuthu Did you get time to review the tests that I've pushed?

Reply:
Hi @privateOmega, sorry for the delay. I’ve been focused a lot on #243 and quite frankly I forgot. I’m taking a break now but will take a look at this next week.